core/opendaq/security/src/permission_manager_impl.cpp

```diff
@@ -0,0 +1,127 @@
+#include <opendaq/permission_manager_impl.h>
+#include <coretypes/impl.h>
+#include <coretypes/string_ptr.h>
+#include <opendaq/user_ptr.h>
+#include <coretypes/validation.h>
+#include <opendaq/permission_config_builder_factory.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+PermissionManagerImpl::PermissionManagerImpl(const PermissionManagerPtr& parent)
+    : children(Dict<IPermissionManager, Bool>())
+    , config(PermissionConfigBuilder().inherit(true).build())
+    , localConfig(PermissionConfigBuilder().inherit(true).build())
+{
+    setParent(parent);
+}
+
+ErrCode INTERFACE_FUNC PermissionManagerImpl::setPermissionConfig(IPermissionConfig* permissionConfig)
+{
+    localConfig = permissionConfig;
+    auto builder = PermissionConfigBuilder();
+
+    if (localConfig.getInherited() && parent.assigned())
+    {
+        const auto parent = getParentManager();
+        const auto parentConfig = parent.getPermissionConfig();
+        builder.inherit(true).extend(parentConfig);
+    }
+
+    builder.extend(localConfig);
+    config = builder.build();
+
+    updateChildPermissions();
+    return OPENDAQ_SUCCESS;
+}
+
+ErrCode INTERFACE_FUNC PermissionManagerImpl::isAuthorized(IUser* user, Permission permission, Bool* authorizedOut)
+{
+    OPENDAQ_PARAM_NOT_NULL(authorizedOut);
+    *authorizedOut = false;
```

Comment:
What happens if a device vendor does not configure the permission manager? We should at some point probably initialize some component to allow everything. I'd maybe set up the permission configuration of components that have a `nullptr` parent to allow everything.

Reply:
I added a default group "everyone" and allow all permissions for it on the root property object.